resolver.go

```diff
@@ -501,6 +544,54 @@ func (r *resolver) forwardQueryStart(w dns.ResponseWriter, msg *dns.Msg, queryID
 	return true
 }
 
+func (r *resolver) addQueryToGC(w dns.ResponseWriter, msg *dns.Msg) {
+	var (
+		cc clientConn
+		ok bool
+	)
+	proto := w.LocalAddr().Network()
+	if proto != "udp" {
+		return
+	}
+
+	r.queryLock.Lock()
+	if cc, ok = r.client[msg.Id]; !ok {
+		return
+	}
+	r.queryLock.Unlock()
+
+	query := sboxQuery{
+		sboxID: r.sb.ID(),
+		dnsID:  msg.Id,
+	}
```

Comment:
Yes, it is make it unique per query so that the lookup is quicker when processing a reply. Right now the individual fields are not used; but having it as a structure will be better if we have to look at the fields.